### PR TITLE
[CL][e2e]: CreateConcentratedPosition

### PR DESCRIPTION
**Pull request by RusAkh**
- CreatePosition
- rename to CreateConcentratedPosition

Closes: #4055

## What is the purpose of the change

This PR adds a function - `CreateConcentratedPosition` - that creates a concentrated position

## Brief Changelog

`CreateConcentratedPosition` added to `commands.go`

## Testing and Verifying

Testing blocked before #4038 is merged

## Documentation and Release Note

no

### CODE REVIEW

**Comment by mattverse** [comment]
Wdyt about pointing this PR to `ruslan/e2e-cl` instead of main?

**Comment by RusAkh** [comment]
@p0mvn Current positions:
Screen Shot 2023-01-27 at 11 15 39

File: tests/e2e/e2e_test.go [hunk 1]
```diff
@@ -49,6 +50,25 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
 	s.Require().Equal(concentratedPool.GetTickSpacing(), tickSpacing)
 	s.Require().Equal(concentratedPool.GetPrecisionFactorAtPriceOne(), sdk.NewInt(precisionFactorAtPriceOne))
 	s.Require().Equal(concentratedPool.GetSwapFee(sdk.Context{}), sdk.MustNewDecFromStr(swapFee))
+
+	// Initialize 2 more nodes
+	node2, err := chainA.GetNodeAtIndex(1)
+	s.Require().NoError(err)
+	node3, err := chainA.GetNodeAtIndex(2)
+	s.Require().NoError(err)
```
Comment (p0mvn) [inline tests/e2e/e2e_test.go:58]: It doesn't matter what node you're submitting it from. I suggest using only one node for simplicity

File: tests/e2e/e2e_test.go [hunk 2]
```diff
@@ -49,6 +50,25 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
...
+
+	minTick, maxTick := cl.GetMinAndMaxTicksFromExponentAtPriceOne(sdk.NewInt(precisionFactorAtPriceOne))
+
+	// Create 2 positions for node1: overlap together, overlap with 2 node3 positions)
+	node1.CreateConcentratedPosition(initialization.ValidatorWalletName, "[-1200]", "400", fmt.Sprintf("1000%s", denom0), fmt.Sprintf("1000%s", denom1), 0, 0, poolID)
```
Comment (p0mvn) [inline tests/e2e/e2e_test.go:63]: What matters is using different addressed to create positions from.

We would like to have positions created from the same addresses as well as some different addresses. Overlapping and non-overlapping

File: tests/e2e/configurer/chain/commands.go [hunk 3]
```diff
@@ -271,6 +282,19 @@ func (n *NodeConfig) CreateWallet(walletName string) string {
 	return walletAddr
 }
 
+func (n *NodeConfig) CreateWalletAndFund(walletName string) string {
```
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:285]: let's specify the coins that we want to fund this with as a parameter to make this more extensible

File: tests/e2e/e2e_test.go [hunk 4]
```diff
@@ -49,6 +51,67 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
...
+
+	// Assert returned positions:
+
+	// assert positions for address1
+	a1p1 := positionsAddress1[0]
+	a1p2 := positionsAddress1[1]
+	// first position first address
+	s.Require().Equal(a1p1.PoolId, poolID)
+	s.Require().Equal(a1p1.LowerTick, int64(-1200))
+	s.Require().Equal(a1p1.UpperTick, int64(400))
```
Comment (p0mvn) [inline tests/e2e/e2e_test.go:91]: let's make a helper called `validateCLPosition` to avoid code duplication

File: tests/e2e/containers/config.go [hunk 5]
```diff
@@ -24,7 +24,7 @@ const (
 	// It should be uploaded to Docker Hub. OSMOSIS_E2E_SKIP_UPGRADE should be unset
 	// for this functionality to be used.
 	previousVersionOsmoRepository = "osmolabs/osmosis-dev"
-	previousVersionOsmoTag        = "v14.x-de392d2b-1674667154"
+	previousVersionOsmoTag        = "v14.x-6b742c84-1675172347"
```
Comment (p0mvn) [inline tests/e2e/containers/config.go:27]: How were you able to push the update? Is it correctly auto-pushing to Docker hub now?

Comment (RusAkh) [reply tests/e2e/containers/config.go:27]: we had https://github.com/osmosis-labs/osmosis/pull/3503 this workflow to push osmosis-dev images

File: tests/e2e/e2e_test.go (hunk 4 above)
Comment (RusAkh) [reply tests/e2e/e2e_test.go:91]: sg

File: tests/e2e/e2e_test.go [hunk 6]
```diff
@@ -49,6 +52,64 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
...
+	positionsAddress2 := node.QueryConcentratedPositions(address2)
+	positionsAddress3 := node.QueryConcentratedPositions(address3)
+
+	// assert number of positions per address
+	s.Require().Equal(len(positionsAddress1), 2)
+	s.Require().Equal(len(positionsAddress2), 1)
+	s.Require().Equal(len(positionsAddress3), 2)
+
+	// Assert returned positions:
+
```
Comment (p0mvn) [inline tests/e2e/e2e_test.go:86]:
```suggestion
```

File: tests/e2e/e2e_test.go [hunk 7]
```diff
@@ -49,6 +52,64 @@ func (s *IntegrationTestSuite) TestConcentratedLiquidity() {
...
+	validateCLPosition := func(position types.FullPositionByOwnerResult, poolId uint64, lowerTick, upperTick int64) {
+		s.Require().Equal(position.PoolId, poolId)
+		s.Require().Equal(position.LowerTick, int64(lowerTick))
+		s.Require().Equal(position.UpperTick, int64(upperTick))
+	}
+
+	// assert positions for address1
+	a1p1 := positionsAddress1[0]
```
Comment (p0mvn) [inline tests/e2e/e2e_test.go:94]: nit: verbosity for readability
```suggestion
	addr1position1 := positionsAddress1[0]
```

**Review by p0mvn — APPROVED** [review]

**Review by p0mvn — APPROVED** [review]

**Pull request merged**